src/pymor/algorithms/timestepping.py

```diff
@@ -69,6 +83,49 @@ def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=N
         -------
         |VectorArray| containing the solution trajectory.
         """
+        num_time_steps = self.estimate_time_step_number(initial_time, end_time)
+        iterator = self.iterate(initial_time, end_time, initial_data, operator, rhs=rhs, mass=mass, mu=mu,
+                                num_values=num_values)
+        U = operator.source.empty(reserve=min(num_time_steps, num_values) if num_values else num_time_steps)
```

Comment:
If `num_values` is specified, shouldn't this always be the value that is used here. If `num_values` is specified, I'd expect this amount of values, even if less time steps are used?

BTW, should we also automatically do linear interpolation for the first-order methods? (Maybe in another PR, however ..)

Reply:
As far as I understand the time-stepping code, it uses `nt` to determine the step size, so `num_values` that is larger than `nt + 1` would still return `nt + 1` steps.

Reply:
If I understand the current code correctly, if `nt` is too small, it repeatedly returns the same value to obtain `num_values`. Whether or not that is desirable behavior is maybe material for another PR. For this one, I'd suggest to stay as close to current behavior as possible.

Reply:
Ok, sounds good.